src/rules/core/id-references-for-requests-rule.js

```diff
@@ -0,0 +1,75 @@
+const Rule = require('../rule');
+const PropertyHelper = require('../../helpers/property');
+const ValidationErrorCategory = require('../../errors/validation-error-category');
+const ValidationErrorSeverity = require('../../errors/validation-error-severity');
+const validationErrorType = require('../../errors/validation-error-type');
+
+/** @typedef {import('../../classes/model-node').ModelNodeType} ModelNode */
+
+class IdReferencesForRequestsRule extends Rule {
+  constructor(options) {
+    super(options);
+    this.targetValidationModes = [
+      'C1Request',
+      'C2Request',
+      'PRequest',
+      'BRequest',
+      'BOrderProposalRequest',
+      'OrderPatch',
+    ];
+    this.targetFields = { OrderItem: ['acceptedOffer', 'orderedItem'] };
```

Comment:
To make this more generically applicable, it would be good to drive this from the data models.

For example we already have this: https://github.com/openactive/data-models/blob/9bc59064c0f0cb822388212cc79049dbdb5c9893/versions/2.x/models/Order.json#L163-L164

```json
      "allowReferencing": true,
```

So would be good to add `referencedFields` alongside these:

https://github.com/openactive/data-models/blob/9bc59064c0f0cb822388212cc79049dbdb5c9893/versions/2.x/models/Order.json#L41-L46

```json
    "orderProposalRequest": {
      "requiredFields": [
        "type",
        "orderProposalVersion"
      ],
      "recommendedFields": [],
```

A good hint for how to do this can be found here: https://github.com/openactive/data-model-validator/blob/64227a084dde40c5836d404a1d7d8d33dbf2dfd1/src/rules/core/required-fields-rule.js#L46